src/main/scala/strawman/collection/Factories.scala

```diff
@@ -22,35 +24,62 @@ trait FromSpecificIterable[-A, +C] extends Any {
   def fromSpecificIterable(it: Iterable[A]): C
 }
 
+/**
+  * Builder factory
+  * @tparam A Element type (e.g. `Int`)
+  * @tparam C Collection type (e.g. `List[Int]`)
+  */
+trait CanBuild[-A, C] {
+  /** Creates a builder */
+  def newBuilder(): Builder[A, C]
+}
+
 /** Base trait for companion objects of unconstrained collection types */
-trait IterableFactory[+CC[_]] extends FromIterable[CC] {
+trait IterableFactory[CC[_]] extends FromIterable[CC] {
   def empty[A]: CC[A]
 
   def apply[A](xs: A*): CC[A] = fromIterable(View.Elems(xs: _*))
 
   def fill[A](n: Int)(elem: => A): CC[A] = fromIterable(View.Fill(n)(elem))
+
+  def newBuilder[A](): Builder[A, CC[A]]
+
+  implicit def canBuild[A]: CanBuild[A, CC[A]] = () => newBuilder[A]()
```

Comment:
Our four flavors of factories (iterable, ordered iterable, map and ordered map) now have a new abstract method `newBuilder` and an implemented implicit member providing a `CanBuild` instance.

Note that the `newBuilder` method is already existing in the current collections, so that’s not really a new thing…

Also note that I added parenthesis to the method, to emphasize the fact that it returns different values each time it is called.